apisix/plugins/jwt-auth.lua

```diff
@@ -357,8 +361,31 @@ local function algorithm_handler(consumer, method_only)
     end
 end
 
+local function set_our_cookie(name, val)
+    core.response.add_header("Set-Cookie", name .. "=" .. val)
+end
+
 
 function _M.rewrite(conf, ctx)
+    local from_header = true
+    local header_key = core.request.header(ctx, conf.header)
+
+    local from_query = true
+
+    if not header_key then
+        from_header = false
+        local uri_args = core.request.get_uri_args(ctx) or {}
+        header_key = uri_args[conf.query]
+        if not header_key then
+            from_query = false
+            local cookie = ctx.var["cookie_" .. conf.cookie]
+            if not cookie then
+                core.log.info("failed to fetch JWT token")
+                return 401, {message = "Missing JWT token in request"}
+            end
+        end
+    end
```

Comment:
we don't need to add this logic in this PR, we just do same things as: https://github.com/apache/apisix/pull/6670/files

Reply:
Sorry, can you provide some description more specific? I mean this logic is similar, but jwt-auth and key-auth is different plugin.

Reply:
The premise of `hide_credentials` is to find out which part of the request JWT exists, header or cookie, we already have the function `fetch_jwt_token` to find JWT.

We can execute `hide_credentials` once the JWT is found in `fetch_jwt_token`.

Reply:
Thank you, I will adjust it.